src/main/java/com/ldtteam/structurize/placement/StructurePlacer.java

```diff
@@ -0,0 +1,358 @@
+            if (storage != null)
+            {
+                storage.addPositionStorage(worldPos, world);
+            }
+
+            final BlockPlacementResult result;
+            switch (operation)
+            {
+                case BLOCK_REMOVAL:
+                    world.removeBlock(worldPos, false);
+                    result = new BlockPlacementResult(localPos, BlockPlacementResult.Result.SUCCESS);
+                    break;
+                case WATER_REMOVAL:
+                    final BlockState worldState = world.getBlockState(worldPos);
+                    if (worldState.getBlock() instanceof IBucketPickupHandler || worldState.getBlock() instanceof FlowingFluidBlock)
+                    {
+                        BlockUtils.removeFluid(world, worldPos);
+                    }
+                    result = new BlockPlacementResult(localPos, BlockPlacementResult.Result.SUCCESS);
+                    break;
+                default:
+                    result = handleBlockPlacement(world, worldPos, storage, localState, handler.getBluePrint().getTileEntityData(worldPos, localPos));
+            }
+            count++;
+
+            if (result.getResult() == BlockPlacementResult.Result.MISSING_ITEMS || result.getResult() == BlockPlacementResult.Result.FAIL || result.getResult() == BlockPlacementResult.Result.BREAK_BLOCK)
+            {
+                return new StructurePhasePlacementResult(localPos, result);
+            }
+        }
+        while (iterateFunction.get() == BlueprintIterator.Result.NEW_BLOCK);
+
+        iterator.reset();
+        return new StructurePhasePlacementResult(BlockPos.ZERO, new BlockPlacementResult(BlockPos.ZERO, BlockPlacementResult.Result.FINISHED));
+    }
+
+    /**
+     * This method handles the block placement.
+     * When we extract this into another mod, we have to override the method.
+     *
+     * @param world          the world.
+     * @param worldPos       the world position.
+     * @param storage        the change storage.
+     * @param localState     the local state.
+     * @param tileEntityData the tileEntity.
+     */
+    public BlockPlacementResult handleBlockPlacement(
+      final World world,
+      final BlockPos worldPos,
+      final ChangeStorage storage,
+      final BlockState localState,
+      final CompoundNBT tileEntityData)
+    {
+        final BlockState worldState = world.getBlockState(worldPos);
+        boolean sameBlockInWorld = false;
+        if (worldState.getBlock() == localState.getBlock())
+        {
+            sameBlockInWorld = true;
+        }
+
+        if (!(worldState.getBlock() instanceof AirBlock))
+        {
+            if (!handler.allowReplace())
+            {
+                return new BlockPlacementResult(worldPos, BlockPlacementResult.Result.BREAK_BLOCK);
+            }
+        }
+
+        for (final CompoundNBT compound : iterator.getBluePrintPositionInfo().getEntities())
```

Comment:
shouldn't this depend on includeEntities?

Reply:
It does, this list is always empty if not includeEntities

Reply:
ah ok